internal/data.go

```diff
@@ -90,28 +92,62 @@ func (x *Timestamp) UnixNano() int64 {
 
 // NewTableData returns a table data type by setting the ts to the current value.
 func NewTableData(data []byte) *TableData {
+	rawData := int32(len(data))
```

Comment:
A nice longer term refactor here is to have one function:

```

func newTableDataFull(data []byte, createdAt *Timestamp, updatedAt *Timestamp, version int32) *TableData {
rawData := int32(len(data))
	return &TableData{
		CreatedAt: createdAt,
		UpdatedAt: updatedAt,
		RawData:   data,
		RawSize:   &rawData,
	}

} 

```
And then the other table data functions call that function to do the actual creation of the TableData.

Reply:
sure